### ISSUE TITLE
Wiremod

### ISSUE THREAD

**Issue by zippy36jr** [wiremod/advdupe2#68]
Since the update the tool has been classifying wiremod stuff such as expression2 chips, gates, pod controllers, etc.. as a "sent" and does not allow players to spawn wire stuff due to the sent limit is set to 0. This was not an issue before the last update.

**Comment by thegrb93** [comment]
Technically chips are SENTs and it is failing because this hook will return false if the max sents its hit  https://github.com/garrynewman/garrysmod/blob/3e138636eb1b0ad6ed785dedf350020755cff5f1/garrysmod/gamemodes/sandbox/gamemode/player.lua#L101

Maybe I can make the hook.Call ignore the gamemode hook and instead use the entity's limit rather than max sents.

**Comment by Divran** [comment]
Or just increase the max sent limit on your server? Why is it set to 0?

**Comment by thegrb93** [comment]
Nevermind the duplicator's entity creation functions checks the limits already. 

I just need to figure out a way to call PlayerSpawnSENT without checking the SENT limit unless it is a SENT which uses sbox_maxsents

**Comment by zippy36jr** [comment]
Its a darkrp server and we don't want players spawning the things in the entities menu, i already blacklist everything and set the limit higher. Still its something that just happened with this update and its kinda annoying because some of the entities use props that people build with.

**Comment by Divran** [comment]
soo you don't allow them to spawn stuff... but you allow them to spawn stuff with adv dupe?
You realize people can just make a dupe in single player and spawn whatever the hell they want right

**Comment by thegrb93** [comment]
@Divran Stuff like hl2 items and spawned sweps should use the sbox_maxsents limit. One problem though, I looked through the code and didn't find any PlayerSpawnSENT hook.Call with the gamemode option, so none of them are actually calling the gamemode PlayerSpawnSENT (thus no limit check). I'm currently trying to find where the limit check is getting hit from.

**Comment by thegrb93** [comment]
I just tested and found that I was able to dupe wire entities with sbox_maxsents 0. Something else must be causing the issue for you.

**Comment by zippy36jr** [comment]
[DupeTest.txt](https://github.com/wiremod/advdupe2/files/40896/DupeTest.txt)

Here is a video showing what is happening: https://www.youtube.com/watch?v=NIPKFOUcyOQ

**Comment by thegrb93** [comment]
Maybe its that entity limit mod you're using.

**Comment by thegrb93** [comment]
I can't remember what its called. What is it?

**Comment by bigdogmat** [comment]
URS I believe

**Comment by thegrb93** [comment]
Yeah it has a limit check inside of the PlayerSpawnSENT hook but for some reason it thinks that wire entities are sents. Try without URS and if it fixes then maybe open an issue for them and link them to this issue.

**Comment by thegrb93** [comment]
Now I need to fix the issue of actual SENTs not checking the sent limit.

**Comment by thegrb93** [comment]
Fixed https://github.com/wiremod/advdupe2/pull/70
